leaf/extensions/tags.py

```diff
@@ -0,0 +1,351 @@
+from bot import LeafBot
+from utils import Paginator
+from discord.ext import commands
+from discord import app_commands
+import discord
+import pytz
+
+from typing import Optional
+
+
+@app_commands.guild_only()
+class TagsCog(commands.GroupCog, name="Tags", group_name="tags"):
+    def __init__(self, bot: LeafBot) -> None:
+        self.bot = bot
+
+    @app_commands.describe(starting_page="The page to start on.")
+    @app_commands.command(name="list", description="Lists all the tags in the server.")
+    async def list_tags(
+        self,
+        interaction: discord.Interaction,
+        starting_page: Optional[int] = 1,
+        silent: Optional[bool] = False,
+    ) -> None:
+        tags = await self.bot.database.fetch(
+            "SELECT * FROM tags WHERE guild_id = $1 AND deleted = FALSE",
+            interaction.guild.id,
+        )
+        embeds = []
+
+        if not tags:
+            # TODO: Remove pagination buttons if there are no tags in the server.
+            embeds.append(
+                discord.Embed(
+                    description="There are no tags in this server.",
+                    color=discord.Color.dark_embed(),
+                )
+            )
+        else:
+            chunks = list(discord.utils.as_chunks(tags, 15))
```

Comment:
This is really nitpicking but list compression is probably more performant (although probably negligibly)

Here is an example of what this might look like:

`chunk_size = 15
num_chunks = (len(tags) + chunk_size - 1) // chunk_size  # calculate number of chunks

for i in range(num_chunks):
    start_index = i * chunk_size
    end_index = min(start_index + chunk_size, len(tags))
    chunk = tags[start_index:end_index]

    embed = discord.Embed(
        description="\n".join([f"• **{tag['name']}** (Uses: {tag['uses']})" for tag in chunk]),
        color=discord.Color.dark_embed(),
    )
    embed.set_footer(text=f"Page {i + 1} / {num_chunks}")
    embeds.append(embed)
`

Reply:
I will have to test how performant it is but that approach seems to sacrifice a lot of readability.